privacyidea/models.py

```diff
@@ -150,6 +152,23 @@ def delete(self):
         return ret
 
 
+# token_container_association_table = (
+#     db.Table('tokencontainertoken',
+#              db.metadata,
+#              db.Column('token_id', db.Integer, db.ForeignKey('token.id'), primary_key=True),
+#              db.Column('container_id', db.Integer, db.ForeignKey('tokencontainer.id'), primary_key=True)))
+#
+
+class TokenContainerToken(MethodsMixin, db.Model):
+    """
+    Association table to link tokens to containers.
+    """
+    __tablename__ = 'tokencontainertoken'
+    __table_args__ = {'mysql_row_format': 'DYNAMIC'}
+    token_id = db.Column('token_id', db.Integer, db.ForeignKey('token.id'), primary_key=True)
+    container_id = db.Column('container_id', db.Integer, db.ForeignKey('tokencontainer.id'), primary_key=True)
```

Comment:
is that allowed? two primary key columns and both foreign keys?
Maybe a unique constraint makes more sense here?

Reply:
Yes that is allowed and serves the same purpose as a unique constraint.
https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#many-to-many